### PR TITLE
feat: [CDS-57810]: Add connector details to timescaledb

### PR DESCRIPTION
**Pull request by RishabhGupta34**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45535)

### CODE REVIEW

**Comment by RishabhGupta34** [comment]
trigger JavaUnitTests8, ut4

**Comment by RishabhGupta34** [comment]
trigger utAll,

File: 120-ng-manager/src/main/resources/timescale/create_connectors.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     org_identifier text  NULL,
+     project_identifier text  NULL,
+     identifier text  NULL,
+     scope text  NULL,
+     name text  NULL,
+     type text  NULL,
+     categories text  NULL,
+     created_by text  NULL,
+     last_updated_by text  NULL,
+     created_at bigint  NULL,
```
Comment (abhinavmittal-harness) [inline 120-ng-manager/src/main/resources/timescale/create_connectors.sql:21]: should we mark created_at as not null

Comment (RishabhGupta34) [reply 120-ng-manager/src/main/resources/timescale/create_connectors.sql:21]: let me add `identifier,name,created_at` as not null

Comment (RishabhGupta34) [reply 120-ng-manager/src/main/resources/timescale/create_connectors.sql]: done

File: 110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.google.inject.Inject;
+
+@OwnedBy(CDP)
+public class ConnectorCDCEntity implements CDCEntity<Connector> {
+  @Inject private ConnectorsChangeDataHandler connectorsChangeDataHandler;
+
+  @Override
+  public ChangeHandler getChangeHandler(String handlerClass) {
+    if (handlerClass.contentEquals("Connectors")) {
```
Comment (sean-harness) [inline 110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java:25]: What is the point of this conditional?

Comment (RishabhGupta34) [reply 110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java:25]: added for future, incase we want to add more handlers.

Comment (RishabhGupta34) [reply 110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java:25]: I can remove it for now

Comment (RishabhGupta34) [reply 110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java]: removed

**Review by sean-harness — COMMENTED** [review]
> Reviewed 3 of 7 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45535)* status: 8 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by sean-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45535#-:-NRXOHTT4BjHhrais4oE:bnfp4nl)* status: 8 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinavmittal-harness — COMMENTED** [review]
> Reviewed 3 of 7 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45535)* status: 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by abhinavmittal-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45535)* status: 1. unresolved discussions.

**Review by abhinavmittal-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45535)* status: 1. unresolved discussions.

**Review by RishabhGupta34 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45535)* status: 1. unresolved discussions. 

___
*[`110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java` line 25 at r5](https://reviewable.io/reviews/harness/harness-core/45535#-NRCw_At7Gbihwb-2PPH-r5-25:-NRXf_wB2Tv79c24pzvY:b3kqx) ([raw file](https://github.com/harness/harness-core/blob/d7021a6f88ceb8c8dd89208f7cb2f2b161b5e0d9/110-change-data-capture/src/main/java/io/harness/entities/ConnectorCDCEntity.java#L25)):*
<details><summary><i>Previously, RishabhGupta34 (Rishabh Gupta) wrote…</i></summary><blockquote>

removed
</blockquote></details>

.

**Pull request merged**